### ISSUE TITLE
Inputs preprocessing

### ISSUE THREAD

**Issue by FrancescoSaverioZuppichini** [facebookresearch/MaskFormer#62]
Hello there!

I am trying to train MaskFormer with some custom data and I am not able to overfit a single batch. After just a few epochs, the model outputs a mask with all zeros. I am definitely doing some wrong :) My preprocessing is the following.

I take as input an image (`channels, height, width`) and a segmentation mask (`height, width`) where each pixel is a label id. Then I convert the segmentation mask to binary masks (`classes, height, width`) where for each class we have `0`s and `1`s. The process is something like this:

```python
if reduce_labels:
      segmentation_map[segmentation_map == 0] = ignore_index
      segmentation_map -= 1
      segmentation_map[segmentation_map == self.ignore_index - 1] = ignore_index
# get all the labels in the image
labels = np.unique(segmentation_map)
# remove ignore index (if we have one)
if ignore_index is not None:
    labels = labels[labels != ignore_index]
# we need to convert mask from [W,H] to [C, W, H]
all_labels = np.arange(num_labels)
# helping broadcast by making mask [1,W,H] and labels [C, 1, 1]
binary_masks = segmentation_map[None] == all_labels[:, None, None]
# convert labels to multi label format
one_hot_labels = np.zeros(num_labels, dtype=np.int64)
one_hot_labels[labels] = 1
return binary_masks.astype(np.float32), one_hot_labels
```
So I feed to MaskFormer a:
- image (`channels, height, width`) 
- binary masks (`classes, height, width`)
- one hot labels (`classes`)

Is this the correct way to preprocess the data for MaskFormer?

Thanks!

Cheers,

Francesco

**Comment by bowenc0221** [comment]
Please refer to the code on how to preprocess data:
1. https://github.com/facebookresearch/MaskFormer/blob/da3e60d85fdeedcb31476b5edd7d328826ce56cc/mask_former/data/dataset_mappers/mask_former_semantic_dataset_mapper.py#L98-L184
2. https://github.com/facebookresearch/MaskFormer/blob/da3e60d85fdeedcb31476b5edd7d328826ce56cc/mask_former/mask_former_model.py#L226-L240

**Comment by FrancescoSaverioZuppichini** [comment]
Thanks for your reply :) I've looked at the code before posting and it is quite mysterious since there is no way to tell what `targets_per_image.gt_classes` is without running the code. I've run the training script and I can see the targets are not one hot encoded, so issues is solved!
